### ISSUE TITLE
Opening and saving assembly changes Disassembly

### ISSUE THREAD

**Issue by C0D3-M4513R** [Col-E/recaf-3x-issues#139]
**Describe the bug**

The disassembly even changes, if you just look at the disassembly, and hit save in a method.

**To Reproduce**

1. Open sample [ftb-library-forge-1802.3.6-build.140.zip](https://github.com/Col-E/Recaf/files/10517177/ftb-library-forge-1802.3.6-build.140.zip) (rename extension to .jar) (source [here ](https://github.com/FTBTeam/FTB-Library/blob/e23e1c07009c8f6233d950c5f8289d0d98a595a8/common/src/main/java/dev/ftb/mods/ftblibrary/snbt/SNBT.java) (possibly not the exact commit, but probably close enough for demo purposes))
2. Navigate to 'dev.ftb.mods.ftblibrary.snbt.SNBT'
3. Look at the logging statements
4. Open the assembly for the method `static SNBTCompoundTag read(Path)`
5. Save
6. Observe the values in the Logging statements disappear in the disassembly

**Screenshots**

```java
    @Nullable
    public static SNBTCompoundTag read(Path path) {
        if (Files.notExists(path, new LinkOption[0]) || Files.isDirectory(path, new LinkOption[0]) || !Files.isReadable(path)) {
            return null;
        }
        try {
            return SNBT.readLines(Files.readAllLines(path, StandardCharsets.UTF_8));
        }
        catch (SNBTSyntaxException ex) {
            FTBLibrary.LOGGER.error("Failed to read " + Platform.getGameFolder().relativize(path) + ": " + ex.getMessage());
            return null;
        }
        catch (Exception ex) {
            FTBLibrary.LOGGER.error("Failed to read " + Platform.getGameFolder().relativize(path) + ": " + ex);
            ex.printStackTrace();
            return null;
        }
    
```
after saving:
```java
    public static SNBTCompoundTag read(Path path) {
        if (Files.notExists(path, new LinkOption[0]) || Files.isDirectory(path, new LinkOption[0]) || !Files.isReadable(path)) {
            return null;
        }
        try {
            return SNBT.readLines(Files.readAllLines(path, StandardCharsets.UTF_8));
        }
        catch (SNBTSyntaxException ex) {
            FTBLibrary.LOGGER.error("Failed to read \\u0001: \\u0001");
            return null;
        }
        catch (Exception ex) {
            FTBLibrary.LOGGER.error("Failed to read \\u0001: \\u0001");
            return null;
        }
    }
```

**Comment by Nowilltolife** [comment]
This has to do with the escaping of the assembler, it tries to escape all escape characters, but probably the arguments for the `invokedynamic StringConcatFactory`, which use \u0001 to mark a replacement character were not escaped correctly.

**Comment by Col-E** [comment]
Addressed in latest, 3x builds as reported in #141
